src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -734,13 +734,15 @@ private EntityHandle GetMemberReferenceHandle(MemberInfo memberInfo)
                 {
                     case FieldInfo field:
                         Type declaringType = field.DeclaringType!;
-                        if (field.DeclaringType!.IsGenericTypeDefinition)
+                        if (declaringType.IsGenericTypeDefinition)
                         {
                             //The type of the field has to be fully instantiated type.
                             declaringType = declaringType.MakeGenericType(declaringType.GetGenericArguments());
                         }
+
                         memberHandle = AddMemberReference(field.Name, GetTypeHandle(declaringType),
-                            MetadataSignatureHelper.GetFieldSignature(field.FieldType, field.GetRequiredCustomModifiers(), field.GetOptionalCustomModifiers(), this));
+                            MetadataSignatureHelper.GetFieldSignature(GetFieldType(field), field.GetRequiredCustomModifiers(), field.GetOptionalCustomModifiers(), this));
```

Comment:
Can this use `GetOriginalMemberIfConstructedType` like the method cases? `GetOriginalMemberIfConstructedType` seems to be doing the same thing as the newly added method.

Reply:
Yes `GetOriginalMemberIfConstructedType` can be used, but need to change the parameter type from `MethodBase` to `MemberInfo`

Reply:
Thanks - done.